### PR TITLE
ci(NODE-4204): add support for latest node version in CI

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

Adds latest node into CI.

##### Is there new documentation needed for these changes?

Nope.

### Double check the following

- [x] Ran `npm run lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/setup-environment.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
```
Comment (nbbeeken) [inline .evergreen/setup-environment.sh:1]:
```suggestion
#! /usr/bin/env bash
```
just for [cross compat reasons](https://stackoverflow.com/questions/16365130/what-is-the-difference-between-usr-bin-env-bash-and-usr-bin-bash)

File: .evergreen/setup-environment.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+PROJECT_DIRECTORY: "$PROJECT_DIRECTORY"
+NODE_VERSION: "$NODE_VERSION"
+PREPARE_SHELL: |
+    set -o errexit
+    set -o xtrace
+    export PROJECT_DIRECTORY="$PROJECT_DIRECTORY"
+    export NODE_VERSION="$NODE_VERSION"
+EOT
+# See what we've done
+cat expansion.yml
```
Comment (nbbeeken) [inline .evergreen/setup-environment.sh:31]: Super minor, can we put the newline at the end here?

File: .evergreen/setup-environment.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export PROJECT_DIRECTORY="$(pwd)"
+
+if [ "$NODE_MAJOR_VERSION" == 'latest' ]; then 
+    NODE_VERSION='node'
+    echo "using latest node version"
+else 
+    # get the latest version of node for given major version
+    NODE_VERSION=$(curl -sL nodejs.org/download/release/latest-v${NODE_MAJOR_VERSION}.x/SHASUMS256.txt -o - | head -n 1 | tr -s ' ' | cut -d' ' -f2 | cut -d- -f2 | cut -dv -f2)
+    echo "LATEST NODE ${NODE_MAJOR_VERSION}.x = $NODE_VERSION"
+fi
```
Comment (nbbeeken) [inline .evergreen/setup-environment.sh:18]: I don't think we need this if stmt, the curl URL can be set to `nodejs.org/download/release/latest/SHASUMS256.txt` then the nvm commands will be using an explicit version number, only for perhaps better logs and just no longer relying on defaults

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: .evergreen/setup-environment.sh (hunk 3 above)
Comment (baileympearson) [reply .evergreen/setup-environment.sh]: per discussion from slack - aligned the download behavior with the driver

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**